library/cartridge_set_control_instance.py

```diff
@@ -15,14 +15,85 @@
 }
 
 
+GET_TWOPHASE_COMMIT_VERSION_TIMEOUT = 60
+
+
+def get_twophase_commit_versions(control_console, advertise_uris):
+    versions, err = control_console.eval_res_err('''
+        local fiber_clock = require('fiber').clock
+        local pool = require('cartridge.pool')
+
+        local uris, timeout = ...
+
+        local deadline = fiber_clock() + timeout
+
+        local connections = {}
+        for i, uri in ipairs(uris) do
+            local conn, err = pool.connect(uri, {
+                wait_connected = false
+            })
```

Comment:
I suggest calling it without options. And removing the second wait_connected loop

Reply:
Fixed.

Reply:
But in this case, it may turn out that the connection will take longer than the timeout, isn't it?